contrib/net/http/httptrace.go

```diff
@@ -0,0 +1,127 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http/httptrace"
+	"time"
+)
+
+type httpTraceResult struct {
+	DNSLookup        time.Duration
+	TCPConnection    time.Duration
+	TLSHandshake     time.Duration
+	ServerProcessing time.Duration
+	NameLookup       time.Duration
+	Connect          time.Duration
+	Pretransfer      time.Duration
+	StartTransfer    time.Duration
+
+	dnsStart      time.Time
+	dnsDone       time.Time
+	tcpStart      time.Time
+	tcpDone       time.Time
+	tlsStart      time.Time
+	tlsDone       time.Time
+	serverStart   time.Time
+	serverDone    time.Time
+	transferStart time.Time
+
+	isTLS    bool
+	isReused bool
+
+	remoteIP   string
+	remotePort string
+}
+
+func WithClientTrace(ctx context.Context, r *httpTraceResult) context.Context {
```

Comment:
Two things:

1. Instead of passing that `struct` from above, can we just pass the `span` and set the tags here directly? Would that work? If you need state, you can just define local variables in this function and make use of its closure to take care of state.
2. Is there any reason this function is exported? I'd like it to be unexported at first, and then if it turns out to be useful for someone (at a later stage in another PR), discuss about its form as part of the public API.

Reply:
This is not resolved.